File: Data Structures & Algorithms/search-2d-matrix/submission-0.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int rows=matrix.length;
        int columns=matrix[0].length;
        for(int i=0;i<rows;i++){
            if(matrix[i][0]<=target && matrix[i][columns-1]>=target){
                int start=0;
                int end=columns-1;
                while(start<=end){
                    int mid=start+(end-start)/2;
                    if(target>matrix[i][mid]){
                        start=mid+1;
                    }else if(target<matrix[i][mid]){
                        end=mid-1;
                    }else{
                        return true;
                    }
                }
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/rotate-matrix/submission-0.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        rows=len(matrix)
        columns=len(matrix[0])
        for r in range(0,rows):
            for c in range(r,columns):
                matrix[r][c],matrix[c][r]=matrix[c][r],matrix[r][c]
        for row in matrix:
            start=0
            end=len(row)-1
            while(start<end):
                row[start],row[end]=row[end],row[start]
                start+=1
                end-=1
    
        
        